backend/src/user/service.ts

```diff
@@ -38,8 +39,16 @@ export const processUserAuthAttempt = async (provider: Provider, profile: passpo
     }
 
     try {
-        const id = await saveUser(user);
-        return { ...user, _id: id } as User;
+        const doesUserExist = await getUserByUuid(user.uuid);
```

Comment:
Tak w sumie to się zastanawiam czy ten fragment kodu to jest wgl potrzebny. Rola jest ustawiana na początku funkcji w obiekcie data. Nie analizowałem szczegółowo co tam się dalej dzieje z tym polem, ale na moje to nie ma takiej potrzeby, żeby sprawdzać czy użytkownik istnieje i ustawiać mu w tym miejscu rolę.

Reply:
Wtedy mi się ciągle cofały uprawnienia z "ADMIN" na "USER" i nie mam pojęcia jak to obejść. Bo nadaję uprawnienia nowym użytkownikom przy pierwszym logowaniu

Reply:
W saveUser można wyrzucić pola których się nie chce updateować tylko przy takim rozwiązaniu nie będzie możliwości zmiany tego z poziomu aplikacji wcale.

Reply:
To trzeba zdecydować, czy będzie chciał to zostawić, żeby np. admini mogli nadawać role z poziomu algodebuga, czy wyrzucamy i będzie trzeba grzebać w bazie. Możecie z Marcinem napisać, który wariant wam bardziej odpowiada

Reply:
Tutaj w sumie pojawia się problem który na pewno powróci w przyszłości czyli, że obecnie nie mamy dobrego podzału na pola usera które odtwarzamy z zewnętrznych danych i nasze lokalne pola. Należałoby rozdzielić pierwsze logowanie od następnych i w przypadku aktualizacji usera zmniejszyć zakres wymaganych pól. To już jest więcej pracy i może być poza scopem tego taska, także jeżeli nie chcesz tego robić, to możesz usunąć przypisanie w funkcji `saveUser` tak jak pisałem wyżej i ja wtedy postaram się to poprawić może przy refaktoryzacji, bo @pawl0wski pewnie też natrafi na ten problem przy ustawianiu lokalnej nazwy użytkownika.

Reply:
Dobra, to można to wrzucić do nowego taska, bo jestem w trakcie maratonu kolosów i nie mam kompletnie czasu na to

Reply:
Dodałem to w https://github.com/Rubix98/algodebug/pull/74